fairlearn/linear_model/_constrained_logistic.py

```diff
@@ -0,0 +1,813 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import numbers
+import warnings
+
+import numpy as np
+import pandas as pd
+from scipy import optimize
+from joblib import Parallel
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model._logistic import (
+    _logistic_loss,
+)
+from sklearn.preprocessing import LabelEncoder, OneHotEncoder
+from sklearn.utils import (
+    check_array,
+    check_consistent_length,
+    compute_class_weight,
+)
+from sklearn.utils.fixes import delayed, _joblib_parallel_args
+from sklearn.utils.multiclass import check_classification_targets
+from sklearn.utils.optimize import _check_optimize_result
+from sklearn.utils.validation import _check_sample_weight
+
+
+# Some helper check functions
+def _check_solver(solver, penalty):
+    all_solvers = ["SLSQP"]
```

Comment:
I've explained my decision for the SLSQP solver briefly in the description of this PR. An alternative is the solver used in #466 (`cvxpy`), but there was some discussion about it regarding problems with its dependencies.

Reply:
Awesome, thanks! I'll resolve this comment then.

Reply:
Why do we only support this solver if it's passed to `scipy.optimize.minimize`? Why not all the other available solvers?

Reply:
`scipy.optimize.minimize` only has three solvers that support constraints, SLSQP, COBYLA and trust-constr. Their default solver for constraints is SLSQP, which is why I started with that one. COBYLA would probably also work fine (I can try this out if we would think that it's better to have it included), since constraints are defined in the same way for SLSQP and COBYLA. trust-constr uses a different way for defining constraints, so that would require some extra code most likely.

Reply:
That makes sense to me, thanks :)